Tests/Utils/Builders/ScenarioAssertionsBuilder.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+namespace OpenDialogAi\Core\Tests\Utils\Builders;
+
+use OpenDialogAi\ActionEngine\Actions\ActionResult;
+use OpenDialogAi\ContextEngine\Contexts\User\UserContext;
+use OpenDialogAi\ContextEngine\Facades\ContextService;
+use OpenDialogAi\ConversationEngine\Reasoners\MessageTemplateMatcher;
+use OpenDialogAi\ConversationEngine\Util\ConversationalState;
+use OpenDialogAi\ConversationEngine\Util\ConversationContextUtil;
+use OpenDialogAi\Core\Conversation\Events\ConversationalState\IncomingIntentStateUpdate;
+use OpenDialogAi\Core\Conversation\Events\ConversationalState\MessageTemplateStateUpdate;
+use OpenDialogAi\Core\Conversation\Events\ConversationalState\OutgoingIntentStateUpdate;
+use OpenDialogAi\Core\Conversation\Facades\ConversationDataClient;
+use OpenDialogAi\Core\ImportExportHelpers\PathSubstitutionHelper;
+use OpenDialogAi\Core\Tests\TestCase;
+
+class ScenarioAssertionsBuilder
+{
+    public TestCase $testCase;
+
+    public bool $isAssertStates = false;
+
+    public function __construct(TestCase $testCase)
+    {
+        $this->testCase = $testCase;
+    }
+
+    /**
+     * @return $this
+     */
+    public function assertGotOneMessage(): ScenarioAssertionsBuilder
+    {
+        $this->testCase->assertCount(1, $this->testCase->messages);
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function assertNoErrors(): ScenarioAssertionsBuilder
+    {
+        foreach ($this->testCase->messages as $message) {
+            $this->testCase->assertNotEquals(MessageTemplateMatcher::ERROR_MESSAGE_ID, $message->getOdId());
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param string $intentOdId
+     * @return $this
+     */
+    public function assertChatAnswer(string $intentOdId): ScenarioAssertionsBuilder
```

Comment:
I think we should rename this as follows:

```suggestion
    public function assertResponseContainsMessage(string $intentOdId): ScenarioAssertionsBuilder
```

Reply:
oki :)